File: test-routing.js
#!/usr/bin/env node

const WebSocket = require('ws');

// Connect to the backend WebSocket
const ws = new WebSocket('ws://localhost:8765/ws');

ws.on('open', () => {
    console.log('Connected to backend WebSocket');
    
    // Test 1: Simple math - should use Direct mode
    console.log('\n📊 Test 1: Simple Math Question');
    console.log('Sending: "what is 2 + 2?"');
    ws.send(JSON.stringify({
        type: 'start_consensus',
        query: 'what is 2 + 2?'
    }));
});

ws.on('message', (data) => {
    try {
        const msg = JSON.parse(data.toString());
        
        if (msg.type === 'a_i_helper_decision') {
            console.log(`\n🎯 AI Helper Routing Decision:`);
            console.log(`   Mode: ${msg.direct_mode ? 'DIRECT (Fast)' : 'CONSENSUS (Full Pipeline)'}`);
            console.log(`   Reason: ${msg.reason}`);
        } else if (msg.type === 'stage_start') {
            console.log(`\n🎯 Stage Started: ${msg.stage}`);
            console.log(`   Model: ${msg.model}`);
        } else if (msg.type === 'token') {
            process.stdout.write(msg.content);
        } else if (msg.type === 'stage_complete') {
            console.log(`\n✅ Stage Complete: ${msg.stage}`);
            console.log(`   Tokens: ${msg.tokens}, Cost: $${msg.cost}`);
        } else if (msg.type === 'consensus_complete') {
            console.log('\n🏁 Consensus Complete!');
            console.log(`   Total Tokens: ${msg.total_tokens}`);
            console.log(`   Total Cost: $${msg.total_cost}`);
            console.log(`   Stages Used: ${msg.stages_used}`);
            
            // After first test, run second test
            if (!ws.secondTestDone) {
                ws.secondTestDone = true;
                setTimeout(() => {
                    console.log('\n\n📊 Test 2: Complex Question');
                    console.log('Sending: "design a full electron app that can use multiple llms"');
                    ws.send(JSON.stringify({
                        type: 'start_consensus',
                        query: 'design a full electron app that can use multiple llms'
                    }));
                }, 2000);
            } else {
                // Both tests done, close connection
                console.log('\n\n✅ All tests complete!');
                ws.close();
                process.exit(0);
            }
        } else if (msg.type === 'error') {
            console.error('\n❌ Error:', msg.message);
            if (msg.details) {
                console.error('Details:', msg.details);
            }
        }
    } catch (e) {
        console.error('Failed to parse message:', e);
    }
});

ws.on('error', (err) => {
    console.error('WebSocket error:', err);
});

ws.on('close', () => {
    console.log('\nWebSocket connection closed');
});

File: test_routing.py
import asyncio
import websockets
import json

async def test_routing():
    uri = "ws://localhost:8765/ws"
    
    # Test queries
    queries = [
        ("What is 2 + 2?", "Simple math - should use Direct mode"),
        ("Explain how the authentication system in this codebase works", "Complex analysis - should use Consensus"),
        ("What is the name of this repository?", "Simple factual - should use Direct"),
        ("Analyze the architecture of this project", "Complex analysis - should use Consensus"),
    ]
    
    for query, expected in queries:
        print(f"\n{'='*60}")
        print(f"Testing: {query}")
        print(f"Expected: {expected}")
        print(f"{'='*60}")
        
        async with websockets.connect(uri) as websocket:
            # Send start consensus message
            message = {
                "type": "start_consensus",
                "query": query,
                "profile": None,
                "conversation_id": None,
                "context": None
            }
            
            await websocket.send(json.dumps(message))
            
            # Listen for AI Helper decision
            while True:
                response = await websocket.recv()
                data = json.loads(response)
                
                if data.get("type") == "ai_helper_decision":
                    print(f"AI Decision: {'Direct Mode' if data['direct_mode'] else 'Consensus Mode'}")
                    print(f"Reason: {data['reason']}")
                    break
                elif data.get("type") == "error":
                    print(f"Error: {data['message']}")
                    break
                elif data.get("type") in ["stage_started", "consensus_complete"]:
                    # Skip other messages for now
                    continue
        
        await asyncio.sleep(1)

asyncio.run(test_routing())
